src/components/GoogleAddressLookup/component.js

```diff
@@ -0,0 +1,293 @@
+                    componentRestrictions: {
+                        country,
+                    },
+                });
+            }
+
+            searchParams = Object.assign(searchParams, otherSearchOptions);
+        }
+        this.setState({ isSearching: true });
+
+        return new Promise((resolve, reject) => {
+            this.autocompleteService.getPlacePredictions(
+                {
+                    ...searchParams,
+                    input: needle,
+                },
+                (predictions, status) => {
+                    this.setState({ isSearching: false });
+                    if (status !== window.google.maps.places.PlacesServiceStatus.OK) {
+                        return reject(status);
+                    }
+                    return resolve(predictions);
+                },
+            );
+        });
+    }
+
+    processSearchResults(results, searchValue) {
+        this.setState({
+            places: results,
+            suggestions: getSuggestions(results, searchValue),
+        });
+    }
+
+    handleError(searchValue) {
+        this.setState({
+            places: [],
+            suggestions: [getSearchForOption(searchValue)],
+        });
+    }
+
+    clear() {
+        this.setState({
+            places: [],
+            suggestions: [],
+        });
+    }
+
+    render() {
+        const {
+            style,
+            label,
+            error,
+            placeholder,
+            disabled,
+            tabIndex,
+            required,
+            id,
+            name,
+            value,
+            hideLabel,
+            onClick,
+            onFocus,
+            onBlur,
+        } = this.props;
+        const { isSearching, suggestions } = this.state;
+        const options = suggestions.length > 0 ? suggestions : null;
+        return (
+            <div id={id} style={style} className={this.getContainerClassName()}>
+                <Lookup
+                    id={this.lookupId}
+                    name={name}
+                    label={label}
+                    debounce
+                    isLoading={isSearching}
+                    placeholder={placeholder}
+                    options={options}
+                    hideLabel={hideLabel}
+                    readOnly={false}
+                    required={required}
+                    tabIndex={tabIndex}
+                    disabled={disabled}
+                    value={getFormattedValue(value, true, <SelectedLocationIcon />)}
+                    onChange={this.handleChange}
+                    onSearch={this.handleSearch}
+                    onClick={onClick}
+                    onFocus={onFocus}
+                    onBlur={onBlur}
+                    error={error}
+                    icon={<LocationIcon />}
+                />
+                {!error && (
+                    <div className="rainbow-google-address-poweredby_container">
```

Comment:
Let's follow the class name naming convention. Call this `rainbow-google-address-lookup_powered-by-google-container`